bundles/org.eclipse.jface/src/org/eclipse/jface/resource/DeferredImageDescriptor.java

```diff
@@ -80,4 +82,12 @@ public Image createImage(boolean returnMissingImageOnError, Device device) {
 		return ImageDescriptor.createFromURL(url).createImage(returnMissingImageOnError, device);
 	}
 
+	@Override
+	public <T> T getAdapter(Class<T> adapter) {
+		if (adapter == URL.class) {
+			return adapter.cast(supplier.get());
```

Comment:
The Supplier should only be queried once for the lifetime of the image descriptor.

Reply:
That's an interesting suggestion, because the whole approach of the DeferredImageDescriptor is to always call the Supplier on demand for the URL and there seems to be no caching intended. Before we do a design change of this class here I would recommend to add a second reviewer that weights his/her words in here.

Reply:
I can only judge from the javadoc that claims:

> Using the {@link ImageDescriptor#createFromURLSupplier(Supplier)} will
> trigger this class, which defers calculating the URL until the image is
> required for the first time.

So it says the calculation of the URL (=the supplier) will be deferred until the image is loaded the first time. It does not claim it is calculated multiple times, what even would defeat the purpose, why should I use a supplier for a "costly" operation when the result is that this costly operation is performed multiple times?

Reply:
If you look at the current usage of `DeferredImageDescriptor`'s internally hold `Supplier<URL>`, you will notice that it is evaluated each time, when image data or an image is evaluated, it seems therefore consistent to do the same within the adapter to URL. If you suggest that the adapter should only return *one* (the first?) URL returned, it would be consistent, if the other functions would do so similarly. This is a design change of thet class. I'm not against it, but I think that should be agreed one by an additional reviewer.

Reply:
Alright, will do that.

Reply:
Done now, to be discussed.